content/en/blog/memory_management/index.md

```diff
@@ -0,0 +1,141 @@
+balloon from one or several VMs can help the host getting some memory back to
+keep the system afloat.
+On the other hand if the host has enough spare memory to give some more to one
+of its VMs, it can deflate the balloon.
+
+I really want to emphasize on the fact that until recently (introduction of
+`virtio-mem`), ballooning was the only reliable way of removing memory from a
+running VM.
+
+#### Hot-plug
+
+This is the proper technique for adding or removing memory to a VM, without
+using a tool like a balloon. And there are two different ways of performing
+memory hot-plug, one is through `ACPI` and the other is with `virtio-mem`.
+
+##### ACPI
+
+This has been the standard way for hot-plugging some memory into a VM for quite
+some time. This is because the mechanism is part of the ACPI specification and
+has been implemented in guest OSes such as Linux and Windows years ago.
+
+Depending on the guest OS, limitations might differ, but for instance with Linux
+there is no way to plug a memory DIMM smaller than 128 MiB. This doesn't allow
+for fine grained hot-plug, but that's good enough for practical use cases, as we
+usually increase a VM's memory using gigabyte as the unit.
+
+The main drawback is the hot-unplug really. This is inconvenient and complicated
+for such operation to succeed because we must remove an entire memory DIMM. For
+instance, if we plug an extra 2 GiB to a running VM, and later on we want to
+remove only 1 GiB, this won't be possible as we can only remove 2 GiB. Of course
+we could have plugged the extra 2 GiB splitting it into 16 DIMMs of 128 MiB, but
+we would hit the limit on how many memory slots are actually supported.
+And even if we could work around this limitation, main part of the complexity
+comes from the fact we have no way to ensure the guest isn't using any of the
+memory slot we're trying to remove.
+This makes hot-unplug a non supported feature with `ACPI` hot-plug mechanism.
+This is the reason why before `virtio-mem` came into the picture, we could
+combine `ACPI` hot-plug with `balloon`, so that we would use the `ACPI`
+mechanism for hot-plugging while the balloon would be used for both hot-unplug
+and fine grained operations.
+
+##### virtio-mem
+
+This fairly recent solution has been introduced to address all use cases we
+mentioned so far. It is a full hot-plug/unplug solution as it doesn't have any
+of the drawbacks from `balloon` and `ACPI` hot-plug. It allows for fine grained
+hot-plug/unplug, relying on the `virtio-mem` driver in the guest to manage small
+chunks of memory. Unless you have strong reasons not to use `virtio-mem` (guest
+kernel not recent enough for instance), this should be the standard way of
+dynamically managing a VM's memory.
+
+### Guest driven
+
+Now that we've covered how the host can manage the guest memory allocated to a
+VM, let's look at the way the guest can notify the host for accurate memory
+management.
+
+#### Free page reporting

```

Comment:
The previous article used title case for the headings. It would be nice to be consistent.